### PR TITLE
docs(spanner): add comment in doc on how to use SPANNER_EMULATOR_HOST

### LINKED ISSUE #4575: spanner: add docs or example for using SPANNER_EMULATOR_HOST
Supersedes: #1617

### PR DESCRIPTION
**Pull request by rahul2393**
Fixes #4575

### CODE REVIEW

File: spanner/examples_test.go [hunk 1]
```diff
@@ -767,3 +772,19 @@ func ExampleNewReadWriteStmtBasedTransaction() {
 		time.Sleep(delay)
 	}
 }
+
+func ExampleClient_usingEmulator() {
+	ctx := context.Background()
+	emulatorAddr := os.Getenv("SPANNER_EMULATOR_HOST")
```
Comment (hengfengli) [inline spanner/examples_test.go:778]: We already have this code inside the client and the users only need to set SPANNER_EMULATOR_HOST: https://github.com/googleapis/google-cloud-go/blob/master/spanner/client.go#L148-L157

Please take a look at this PR as a reference: https://github.com/googleapis/google-cloud-go/pull/4656/files

**Review by hengfengli — APPROVED** [review]
LGTM.

**Comment by hengfengli** [comment]
@codyoss Does this look good to you?

**Comment by codyoss** [comment]
@hengfengli Looks good to me, thanks for working on this!

**Pull request merged**